File: mse-report/src/main/java/upeu/mse_report/dto/ReportDTO.java
package upeu.mse_report.dto;

import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class ReportDTO {
    private Long idReport;
    private String generatedBy;
    private String type;
    private String format;
    private String status;

    private LocalDateTime createdAt;
    private LocalDateTime generatedAt;

    private String fileUrl;

    private List<ReportLogDTO> logs;
}

File: matdash-free-angular-v3/main/src/app/models/report/ReportDTO.ts
import { ReportType, ReportFormat, ReportStatus } from './report.enums';
import { ReportLogDTO } from './ReportLogDTO';

export interface ReportDTO {
  idReport: number;
  generatedBy: string;
  type: ReportType;
  format: ReportFormat;
  status: ReportStatus;

  createdAt: string;
  generatedAt: string;

  fileUrl: string | null;

  logs: ReportLogDTO[];
}
